### ISSUE TITLE
Many fields not available in (REPL) autocomplete

### ISSUE THREAD

**Issue by fonsp** [JuliaWeb/HTTP.jl#660]
If I type `HTTP.UR` in the Julia REPL and press TAB, nothing comes up. This can make HTTP.jl difficult to use.

The same for:
`URIs`, `URI`, `Router`, `@register`, `splitpath`, `unescapeuri`, `queryparams`, and more

**Comment by chris-b1** [comment]
I think something like https://github.com/JuliaLang/julia/issues/38162 would be necessary to fix this.  It proposes "scoped" exports, so that names like `Router` could marked as API without being brought in as names by `using`

**Comment by fonsp** [comment]
I think that this is something different. Normally autocomplete also shows fields that are not exported:
```julia
julia> module Example
       x1 = 1
       x2 = 2
       export x1
       end
Main.Example

julia> using .Example

julia> Example.x

x1 x2
```

**Comment by quinnj** [comment]
@fonsp, could you post an update here? Is there something actionable for HTTP.jl or URIs.jl here? I'm happy to dig in, but just want to know if you have a better idea of the context/issue here and what we would need to do and where.

**Comment by quinnj** [comment]
I'm not sure there's anything we can do here in HTTP.jl apart from getting rid of all submodules and just defining everything in a single HTTP module, but that seems quite heavy-handed and gives up a lot of clean separation we have.

**Comment by fredrikekre** [comment]
Could make everything that is public API available in the HTTP module, and the rest in an Internals module. Doesn't have to mean code has to be organized that much, see e.g. https://github.com/fredrikekre/Canvas.jl

**Comment by quinnj** [comment]
That's true and actually probably wouldn't be too hard. I think we would just want to look at the existing submodules and what they export and then define those names at the top-level HTTP module, then wrap everything else in an Internals module. Or maybe we don't even need the Internals module, since we already have all these submodules where they can just import the public API methods/structs to provide the functionality.

Anyone want to have a go at implementing this? If not, I can get around to it in the next little while.

**Comment by quinnj** [comment]
Checking on this again in the 2.0 release, I think everything is resolved!
